### PR TITLE
Move common remote options to generic

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

File: pulpcore/cli/common/context.py [hunk 1]
```diff
@@ -319,7 +321,11 @@ def __init__(
...
         return {
-            key: value.pulp_href if isinstance(value, PulpEntityContext) else value
+            key: None
+            if key in self.NULLABLES and value == ""
+            else value.pulp_href
+            if isinstance(value, PulpEntityContext)
+            else value
             for key, value in body.items()
             if value is not None
         }
```
Comment (daviddavis) [inline pulpcore/cli/common/context.py:331]: I'm finding this code hard to parse but maybe it's because I don't often see ternaries in Python?

Comment (mdellweg) [reply pulpcore/cli/common/context.py:331]: Is this a request to rewrite it?
I could turn it into a loop (which i don't want) or rewrite the quaternary in to a helper function (not really keen, but better).

Comment (daviddavis) [reply pulpcore/cli/common/context.py:331]: A nested ternary in a dict comprehension seems very hard to read. What's the downside to using a loop?

Comment (mdellweg) [reply pulpcore/cli/common/context.py:331]: Would look like this:
```
    def _preprocess_value(self, key: str, value: Any) -> Any
            if key in self.NULLABLES and value == "":
                return None
            if isinstance(value, PulpEntityContext):
                return value.pulp_href
            return value

    def preprocess_body(self, body: EntityDefinition) -> EntityDefinition:
        return {
            key: self._preprocess_value(key, value)
            for key, value in body.items()
            if value is not None
        }
```

**Review by daviddavis — APPROVED** [review]
